packages/react/src/framework-delegate.tsx

```diff
@@ -22,17 +22,19 @@ export const ReactDelegate = (
     const componentWithProps = component(propsOrDataObj);
     const hostComponent = createPortal(componentWithProps, div);
 
-    refMap.set(component, hostComponent);
+    refMap.set(div, hostComponent);
 
     addView(hostComponent);
 
     return Promise.resolve(div);
   };
 
-  const removeViewFromDom = (_container: any, component: ReactComponent): Promise<void> => {
+  const removeViewFromDom = (_container: any, component: HTMLElement): Promise<void> => {
     const hostComponent = refMap.get(component);
     hostComponent && removeView(hostComponent);
 
+    component.remove();
```

Comment:
Are there any considerations we need to make with conditionally rendering pages similar to https://github.com/ionic-team/ionic-framework/pull/25663?

Reply:
I do not believe the same considerations apply for `IonNav`. With `IonModal` the React node is relocated in the DOM and this causes potential issues when the surrounding React nodes are added or removed (causing React to incorrect remove the modal reference - which is why we render the `template` buffer elements). With `IonNav` the React node is stable and the components are added/removed as children of the `IonNav`.